src/protocolsupport/protocol/typeremapper/pe/PEDataValues.java

```diff
@@ -140,7 +141,12 @@ private static void registerLivingType(NetworkEntityType type, int networkId) {
 	}
 
 	public static int getEntityTypeId(NetworkEntityType type) {
-		return entityType.get(type);
+		try {
+			return entityType.get(type);
+		} catch(NullPointerException e) {
+			System.err.println("Missing entity mapping type for " + type);
```

Comment:
is there any good logger available to use? feel like these are important and should stick around, but probably not like this

Reply:
If there's an actual missing entity we should implement it, not keep a log on every server. So I think its fine for now. I used to let it crash the connection to force me to implement the entity maps lol.

Reply:
catch*space*( ?

Reply:
ill add the illusioner here

Reply:
I added something similar for missing bounding boxes. While a crash is "good" in that it forces you to fix it, it's also irritating if you're working on something else. Nevertheless, even in production, I prefer to have the issue logged and some reasonable value substituted, over a crash. It's more "graceful degradation", allowing people to still use the plugin even if we've missed something.

However, we should (somehow) be sure we really check for these to happen before declaring alpha/beta/whatever. Not sure of the best way to remember to do that...

Reply:
We should write a test. Let it fail on build if entity has no mapping.